auth_proxy/services/oauth.py

```diff
@@ -23,17 +31,32 @@ def __init__(self, db, oauth):
         oauth.tokengetter(self.cb_tokengetter)
         oauth.tokensetter(self.cb_tokensetter)
 
-    def register(self, client_id, redirect_uris, scopes, client_name, client_secret=None):
-        """ Register new clients.
+    def register(self, redirect_uris, scopes, client_name=None):
+        """ Register new clients. See RFC7591.
         """
-        if client_secret is None:
-            client_secret = str(uuid.uuid4())
+
+        if not redirect_uris:
+            raise OAuthServiceError(
+                'invalid_client_metadata',
+                'One or more redirect URIs are required.'
+            )
+
+        for uri in redirect_uris:
+            validate_redirect_uri(uri)
+
+        # TODO: validate scopes?
+
+        client_id = str(uuid.uuid4())
+        client_secret = str(uuid.uuid4())
```

Comment:
A bit beside the point, but while we're in here... https://stackoverflow.com/a/43219729/318206 is a better way to generate a `client_secret` (just from a best practices perspective; obviously this particular system isn't actually meant to be secure).

Reply:
I wasn't aware of the `secrets` module - very cool!